script/testing/util/db_server.py

```diff
@@ -33,134 +49,170 @@ def __init__(self, host=DEFAULT_DB_HOST, port=DEFAULT_DB_PORT, build_type='', se
         self.db_process = None
 
     def run_db(self, is_dry_run=False):
-        """ Start the DB server """
-        # Allow ourselves to try to restart the DBMS multiple times
-        attempt_to_start_time = time.perf_counter()
+        """
+        Start the DBMS.
+
+        Parameters
+        ----------
+        is_dry_run : bool
+            True if the commands that will be run should be printed,
+            with the commands themselves not actually executing.
+
+        Returns
+        -------
+        True if the database was started successfully and false otherwise.
+
+        Warnings
+        --------
+        The NoisePage DBMS MUST have logging enabled as it is the stdout output
+        which is used to determine whether the database is ready to accept
+        incoming connections.
+        """
         server_args_str = generate_server_args_str(self.server_args)
         db_run_command = f'{self.build_path} {server_args_str}'
+
         if is_dry_run:
-            LOG.info(f'Server start command: {db_run_command}')
-            return
-        for attempt in range(DB_START_ATTEMPTS):
```

Comment:
Interesting. You removed the multiple attempts to start the DB and the exponential backoff. Any reason?

Reply:
If it doesn't work, something's wrong with the setup, and I would rather it fails as fast as possible than trying to flake through.